lib/db_entry.go

```diff
@@ -0,0 +1,75 @@
+package lib
+
+import "bytes"
+
+// -------------------------------------------------------------------------------------
+// DBEntry
+// -------------------------------------------------------------------------------------
+
+// DBEntry is used to represent a database record. It's more convenient than passing
+// <key, value> everywhere.
+type DBEntry struct {
```

Comment:
Didn't we have this defined somewhere already like for Hypersync or something? I swear I've seen it before.

Reply:
Haha it was, I just moved it around.